artifactory_cleanup/artifactorycleanup.py

```diff
@@ -54,9 +59,12 @@ def cleanup(self, block_ctx_mgr, test_ctx_mgr) -> Iterator[CleanupSummary]:
                 print(f"Found {len(artifacts_to_remove)} artifacts AFTER filtering")
 
                 # Delete artifacts
-                for artifact in artifacts_to_remove:
-                    with test_ctx_mgr(get_name_for_ci(artifact)):
-                        policy.delete(artifact, destroy=self.destroy)
+                with ThreadPoolExecutor(max_workers=int(self.worker_count)) as executor:
+                    for artifact in artifacts_to_remove:
+                        with test_ctx_mgr(get_name_for_ci(artifact)):
+                            executor.submit(
+                                policy.delete, artifact, destroy=self.destroy, ignore_not_found=self.ignore_not_found
+                            )
```

Comment:
Does it get output right? because `test_ctx_mgr` prints some additional lines and after that we commit it to thread pool - it means that it can (and probably will) print lines in wrong order.
We need to move the context `test_ctx_mgr` inside the block

Reply:
inside which block? currently i just added the ThredPoolExecutor block above your for loop

Reply:
we need to move `test_ctx_mgr` inside ` executor.submit`, so it'll execute in the same thread as "delete" api call